src/Block/Element/AbstractBlock.php

```diff
@@ -84,75 +74,35 @@ public function __construct()
      */
     public function getParent()
     {
-        return $this->parent;
+        return parent::getParent();
```

Comment:
If we implement the `ContainerNode` idea then perhaps we can make the `Node::getParent()` docblock be `@return ContainerNode`.